tools/rosclean/src/rosclean/__init__.py

```diff
@@ -134,17 +135,28 @@ def get_disk_usage(d):
     :raises: :exc:`CleanupException` If get_disk_usage() cannot be used on this platform
     """
     # only implemented on Linux and FreeBSD for now. Should work on OS X but need to verify first (du is not identical)
-    if platform.system() == 'Linux':
+    cmd = []
+    unit = 1
+    du = find_executable('du')
+    if du != None:
+        if platform.system() == 'Linux':
+            cmd = [du, '-sb', d]
+        elif platform.system() == 'FreeBSD':
+            cmd = [du, '-skA', d]
+            unit = 1024
         try:
-            return int(subprocess.Popen(['du', '-sb', d], stdout=subprocess.PIPE).communicate()[0].split()[0])
+            # detect BusyBox du command by following symlink
+            if os.path.basename(os.readlink(du)) == 'busybox':
+                cmd = [du, '-sk', d]
+                unit = 1024
```

Comment:
What exception do you expect to be raise in this block? Please catch only the expected types rather than any.

Reply:
`readlink` raises `OSError` if the target is not symlink. I'll fix it to catch only `OSError`.